support-frontend/assets/helpers/redux/checkout/payment/contributionsSideEffects.ts

```diff
@@ -11,4 +18,13 @@ export function addPaymentsSideEffects(
 			storage.setSession('selectedPaymentMethod', action.payload.paymentMethod);
 		},
 	});
+
+	startListening({
+		actionCreator: setPaymentMethodCountryAndState,
+		effect(action, listenerApi) {
+			listenerApi.dispatch(setBillingCountry(action.payload[0]));
+			action.payload[1] &&
+				listenerApi.dispatch(setBillingState(action.payload[1]));
+		},
+	});
```

Comment:
If we do need to retain the separate `setPaymentMethodCountryAndState` action, I would suggest that rather than using a listener we use `extraReducers` instead, since all we're doing here is setting more bits of Redux state. cf. the `extraReducer` for the `validateForm` action: https://github.com/guardian/support-frontend/blob/eb4fa9e84801fdd8ddbf69a6adc93d9125aa8487/support-frontend/assets/helpers/redux/checkout/address/reducer.ts#L90

Reply:
The plan now is to go with the second approach and get rid of the separate payment method country and state!